fluent-package/templates/package-scripts/fluent-package/deb/postinst

```diff
@@ -22,38 +20,113 @@ add_system_user() {
-    mkdir -p /etc/<%= compat_package_dir %>/plugin
-    mkdir -p /var/log/<%= compat_package_dir %>
+    mkdir -p /etc/<%= package_dir %>
+    mkdir -p /etc/<%= package_dir %>/plugin
+    mkdir -p /var/log/<%= package_dir %>
+    if [ -d /etc/<%= compat_package_dir %> -a ! -h /etc/<%= compat_package_dir %> ]; then
+	touch $v4migration
+	# /etc/<%= compat_package_dir %> migration from v4
+	if [ -d /etc/<%= compat_package_dir %>/plugin -a -n "$(ls /etc/<%= compat_package_dir %>/plugin)" ]; then
+	    echo "Migrating from /etc/<%= compat_service_name %>/plugin/ to /etc/<%= package_dir %>/plugin/..."
+	    mv -f /etc/<%= compat_service_name %>/plugin/* /etc/<%= package_dir %>/plugin/
+	fi
+	if [ -f /etc/<%= compat_package_dir %>/<%= service_name %>.conf ]; then
+	    # Avoid conflict with packaged version
+	    echo "#"
+	    echo "# Migrating from /etc/<%= compat_package_dir %>/<%= service_name %>.conf to /etc/<%= package_dir %>/<%= service_name %>.conf.rpmmoved"
+	    echo "#"
+	    mv -f /etc/<%= compat_package_dir %>/<%= service_name %>.conf /etc/<%= package_dir %>/<%= service_name %>.conf.dpkg-old
+	fi
+	if [ -f /etc/<%= compat_package_dir %>/<%= compat_service_name %>.conf ]; then
+	    echo "Migrating from /etc/<%= compat_package_dir %>/<%= compat_service_name %>.conf to /etc/<%= package_dir %>/<%= compat_service_name %>.conf"
+	    cp -f /etc/<%= compat_package_dir %>/<%= compat_service_name %>.conf /etc/<%= package_dir %>/<%= compat_service_name %>.conf
+	    for d in $(ls /etc/<%= compat_package_dir %>); do
+		if [ ! "$d" = "plugin" -a ! "$d" = "<%= compat_service_name %>.conf" ]; then
+		    # /etc/<%= compat_package_dir %>/plugin will be removed from td-agent
+		    mv -f /etc/<%= compat_package_dir %>/$d /etc/<%= package_dir %>/
+		fi
+	    done
+	    echo "Refer previous configuration <%= compat_service_name %>.conf ..."
+	    sed -i"" /etc/default/<%= service_name %> -e "/FLUENT_CONF/c FLUENT_CONF=/etc/<%= package_dir %>/<%= compat_service_name %>.conf"
+	fi
+    fi
+    if [ -h /etc/systemd/system/td-agent.service ]; then
+	systemctl is-active <%= compat_service_name %> >/dev/null
+	if [ $? -eq 0 ]; then
+	    # Want to restart with new user/group here,
+	    # but to avoid holding file descriptor under /var/log/<%= compat_package_dir %>/,
+	    # delay restarting <%= service_name %> service.
+	    touch $v4migration_with_restart
+	fi
+    fi
+    if [ -d /var/log/<%= compat_package_dir %> -a ! -h /var/log/<%= compat_package_dir %> ]; then
+	# /var/log/<%= compat_package_dir %> migration from v4
+	if [ -d /var/log/<%= compat_package_dir %>/buffer ]; then
+	    if [ -n "$(ls /var/log/<%= compat_package_dir %>/buffer)" ]; then
+		mv -f /var/log/<%= compat_package_dir %>/buffer/* /var/log/<%= package_dir %>/buffer/
+	    fi
+	fi
+	for d in $(ls /var/log/<%= compat_package_dir %>); do
+	    if [ ! "$d" = "buffer" ]; then
+		# /var/log/<%= compat_package_dir %>/buffer will be removed from td-agent
+		mv -f $d /var/log/<%= package_dir %>/
```

Comment:
https://github.com/fluent/fluent-package-builder/pull/489#issuecomment-1584326580

> Especially, the migration of /var/log/td-agent/td-agent.log doesn't work correctly.
> The old log file is completely disappeared.

I found the cause.

```suggestion
		mv -f /var/log/<%= compat_package_dir %>/$d /var/log/<%= package_dir %>/
```

Reply:
Thanks.